docs/madr/decisions/022-retry-host-selection.md

```diff
@@ -0,0 +1,75 @@
+# Retry Host Selection
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/5897
+
+## Context and Problem Statement
+
+We want to add some additional retry fields plumbed through from Envoy to our Retry policy based on user demand. This is an HTTP only policy (no GRPC).
+
+```
+retry_priority
+retry_host_predicate
+host_selection_retry_max_attempts
+```
+
+### Design 
+
+Envoy has "host predicates" and "priority predicates" to enforce an additional logic 
+when selecting a host for the attempt. With `retry_host_predicate` we get:
+
+* don't retry on previously retried hosts
+* don't retry on canary hosts (the host should be preliminarily marked with `canary: true`)
```

Comment:
do you mean Kuma tag `canary: true`? or how are they marked as canary?

Reply:
It looks like it's a tag on the lbEndpoint in Envoy config. If we never set this for any reason this option can probably be ignored / removed from this initial implementation.

Reply:
We don't so I think we can omit this for initial implementation